packages/admin-ui/src/number-input/number-input.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  FormControl,
+  FormControlLabel,
+  FormControlMessage,
+} from '../form-control'
+
+import { useNumberInput } from './number-input.state'
+import { useMessageFormatter } from '../i18n'
+
+import { messages } from '../form-control/form-control.i18n'
+
+import * as style from './number-input.style'
+
+export const NumberInput = forwardRef(
+  (props: NumberInputProps, ref: Ref<HTMLInputElement>) => {
+    const {
+      prefix,
+      suffix,
+      disabled,
+      optional,
+      error,
+      errorText,
+      helpText,
+      label,
+      id: defaultId,
+      value: currentValue,
+      step,
+      onChange,
+      ...inputProps
+    } = props
+
+    const { min, max, placeholder } = inputProps
+
+    const { handleChange, handleDecrement, handleIncrement } = useNumberInput({
+      currentValue,
+      step,
+      min,
+      max,
+      onChange,
+    })
+
+    const id = useId(defaultId)
+
+    const formatMessage = useMessageFormatter(messages.formControl)
+    const optionalPlaceholder =
+      !label && optional ? formatMessage('optional') : ''
+
+    return (
+      <FormControl error={error} optional={optional}>
+        {label && <FormControlLabel htmlFor={id}>{label}</FormControlLabel>}
+        <InputContainer error={error} disabled={disabled} csx={style.container}>
+          {prefix && <InputTerm type="prefix">{prefix}</InputTerm>}
+          <InputElement
+            ref={ref}
+            type="number"
+            id={id}
+            value={currentValue}
+            step={step}
+            placeholder={optionalPlaceholder ?? placeholder}
+            onChange={handleChange}
+            disabled={disabled}
+            csx={style.input}
+            {...inputProps}
+          />
+          <Button
+            aria-label="Decrement"
+            variant="tertiary"
+            icon={<IconCaretDown />}
+            onClick={handleDecrement}
+            disabled={disabled || (currentValue && currentValue <= min)}
+            csx={style.spinButton}
+          />
+          <Button
+            aria-label="Increment"
+            variant="tertiary"
+            icon={<IconCaretUp />}
+            onClick={handleIncrement}
+            disabled={disabled || (currentValue && currentValue >= max)}
+            csx={style.incrementButton}
+          />
+          {suffix && <InputTerm type="suffix">{suffix}</InputTerm>}
+        </InputContainer>
+        <FormControlMessage helpText={helpText} errorText={errorText} />
+      </FormControl>
+    )
+  }
+)
+
+NumberInput.displayName = 'NumberInput'
+
+type JSXInputProps = ComponentPropsWithoutRef<'input'>
+
+interface NumberInputOptions {
+  prefix?: ReactNode
+  suffix?: ReactNode
+  label?: ReactNode
+  value: number | string
+  step: number
+  min: number
+  max: number
+  optional?: boolean
+  error?: boolean
+  errorText?: ReactNode
+  helpText?: ReactNode
+  onChange: (value: number | string) => void
```

Comment:
I am wondering if this "custom" onChange make sense, because the input value will be change by typing and by clicking the increment and decrement buttons of this component 🧐

Reply:
I guess it doens't really make a difference if we are only accepting number inputs (even if they are converted from strings). Should it be required though?

Reply:
I guess it doens't really make a difference if we are only accepting number inputs (even if they are converted from strings). Should it be required though?